From c1cf1500b9a07b962abc93b1871863c0809400dc Mon Sep 17 00:00:00 2001
From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Fri, 10 Jun 2016 16:47:02 +0200
Subject: [PATCH] mediamanager: fix invalid memory read

Loop was accessing 'next' pointer in already deallocated GList item.
Reported by Valgrind.
---
 libpurple/mediamanager.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libpurple/mediamanager.c b/libpurple/mediamanager.c
index 80c27b1..46d2020 100644
--- a/libpurple/mediamanager.c
+++ b/libpurple/mediamanager.c
@@ -493,14 +493,18 @@ purple_media_manager_remove_media(PurpleMediaManager *manager, PurpleMedia *medi
 
 #ifdef HAVE_MEDIA_APPLICATION
 		g_mutex_lock (&manager->priv->appdata_mutex);
-		for (list = manager->priv->appdata_info; list; list = list->next) {
+		list = manager->priv->appdata_info;
+		while (list) {
 			PurpleMediaAppDataInfo *info = list->data;
+			GList *next = list->next;
 
 			if (info->media == media) {
 				manager->priv->appdata_info = g_list_delete_link (
 					manager->priv->appdata_info, list);
 				free_appdata_info_locked (info);
 			}
+
+			list = next;
 		}
 		g_mutex_unlock (&manager->priv->appdata_mutex);
 #endif
-- 
2.7.4

